### PR TITLE
Handle edge case values for abstol and span with ITP

### PR DESCRIPTION
**Pull request by dcourteville**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Allows ITP to handle very small `abstol` (and in particular `abstol=0.0`) and very big intervals. This is required to use it in DiffEqBase.

### CODE REVIEW

File: lib/BracketingNonlinearSolve/src/itp.jl [hunk 1]
```diff
@@ -96,8 +96,13 @@ function SciMLBase.__solve(
...
     k1 = alg.scaled_k1 * span^(1 - k2) # k1 > 0
     n0 = alg.n0
-    n_h = exponent(span / (2 * ϵ))
-    ϵ_s = ϵ * exp2(n_h + n0)
+    if span / 2 > ϵ * floatmax(typeof(span))
+        # Workaround for when span / (2 * ϵ) == Inf
+        ϵ_s = span / 2 * exp2(n0)
+    else
+        n_h = exponent(span / (2 * ϵ))
+        ϵ_s = ϵ * exp2(n_h) * exp2(n0)
```
Comment (oscardssmith) [inline lib/BracketingNonlinearSolve/src/itp.jl:104]: why isn't this still `exp2(n_h+n0)`? I would expect that to be faster.

**Comment by oscardssmith** [comment]
yeah this makes sense. It is somewhat annoying that this is parameterized a little bit inefficiently, but realistically this won't cost any real performance.
